<a href="https://colab.research.google.com/github/katherinarroyaveprogramacion/lenguaje_deprogramacion/blob/main/taller_portafolio_inversion.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# 📊 Taller Práctico: Análisis de un Portafolio de Inversión
## Lenguajes de Programación – Ingeniería Financiera

En este taller practicarás el uso de **tuplas, listas y diccionarios** para analizar un portafolio de inversión.

### Objetivos
- Usar **tuplas** para información fija
- Usar **listas** para almacenar datos
- Usar **diccionarios** para relacionar información
- Practicar **indexación, append() y len()**
- Analizar un portafolio de inversión



## Parte 1 — Tuplas (Información fija)
Las tuplas son útiles cuando los datos **no cambian**.

In [ ]:
empresas = ("Apple", "Microsoft", "Amazon", "Tesla", "Google")
print(empresas)

('Apple', 'Microsoft', 'Amazon', 'Tesla', 'Google')


### Ejercicio
Imprime:
- La primera empresa
- La tercera empresa
- La última empresa

In [ ]:
# Escribe tu código aquí
print(empresas[0])
print(empresas[2])
print(empresas[4])


Apple
Amazon
Google


## Parte 2 — Lista (Precio de las acciones)

In [ ]:
precios_acciones = [185, 320, 145, 250, 275]
print(precios_acciones)

[185, 320, 145, 250, 275]


### Ejercicio
Imprime:
- Precio de Apple
- Precio de Tesla
- Cantidad de empresas en el portafolio usando len()

In [ ]:
# Escribe tu código aquí
print(precios_acciones[0])
print(precios_acciones[3])
print(len(precios_acciones))


185
250
5


## Parte 3 — Diccionario (Cantidad de acciones)

In [ ]:
acciones_portafolio = {
    "Apple": 10,
    "Microsoft": 5,
    "Amazon": 8,
    "Tesla": 4,
    "Google": 6
}
print(acciones_portafolio)

{'Apple': 10, 'Microsoft': 5, 'Amazon': 8, 'Tesla': 4, 'Google': 6}


### Ejercicio
Imprime:
- Acciones de Apple
- Acciones de Google

In [ ]:
# Escribe tu código aquí
print(acciones_portafolio["Apple"])
print(acciones_portafolio["Google"])


10
6


## Parte 4 — Valor invertido

In [ ]:
valor_apple = precios_acciones[0] * acciones_portafolio["Apple"]
print(valor_apple)

1850


### Ejercicio
Calcula el valor invertido en Microsoft, Amazon, Tesla y Google.

In [ ]:
# Escribe tu código aquí
valor_microsoft = precios_acciones[1]*acciones_portafolio["Microsoft"]
valor_amazon = precios_acciones[2]*acciones_portafolio["Amazon"]

valor_tesla = precios_acciones[3]*acciones_portafolio["Tesla"]
valor_google = precios_acciones[4]*acciones_portafolio["Google"]
print(valor_microsoft)
print(valor_amazon)
print(valor_tesla)
print(valor_google)


1600
1160
1000
1650


## Parte 5 — Lista del portafolio

In [ ]:
valor_portafolio = []
valor_portafolio.append(valor_apple)
# Agrega los demás valores calculados
valor_portafolio
valor_portafolio.append(valor_microsoft)
valor_portafolio.append(valor_amazon)
valor_portafolio.append(valor_tesla)
valor_portafolio.append(valor_google)
print(valor_portafolio)



[1850, 1600, 1160, 1000, 1650]


In [ ]:
print("Valor total del portafolio:", sum(valor_portafolio))
print("Promedio:", sum(valor_portafolio)/len(valor_portafolio))
print("Número de empresas:", len(valor_portafolio))
print("Valor más alto:", max(valor_portafolio))
print("Valor más bajo:", min(valor_portafolio))



Valor total del portafolio: 7260
Promedio: 1452.0
Número de empresas: 5
Valor más alto: 1850
Valor más bajo: 1000


## Reflexión
1. ¿En qué empresa hay mayor inversión?
#La mayor inversión está en Apple, con un valor de 1850, siendo el monto más alto del portafolio.
2. ¿El portafolio está diversificado?
#Sí, el portafolio está diversificado porque la inversión está distribuida entre cinco empresas diferentes:
Apple, Microsoft, Amazon, Tesla, Inc. y Google.

3. ¿Qué pasaría si Tesla sube 10%?
 #Esto haría que el valor total del portafolio aumente.

4. ¿Por qué son útiles las estructuras de datos en finanzas?
 #Las estructuras de datos (tuplas, listas y diccionarios) permiten organizar y analizar información financiera como precios de acciones, cantidades y valores invertidos.Gracias a ellas se pueden calcular rápidamente el valor total del portafolio, promedios, máximos y mínimos, lo que ayuda a tomar mejores decisiones de inversión.